pkg/member/member_control.go

```diff
@@ -136,6 +144,34 @@ func (m *memberControl) AddMemberAsLearner(ctx context.Context) error {
 	return nil
 }
 
+// AddLearnerWithRetry add a new member as a learner with exponential backoff.
+func (m *memberControl) AddLearnerWithRetry(ctx context.Context, retrySteps int, dataDir string) error {
+	backoff := miscellaneous.CreateBackoff(RetryPeriod, retrySteps)
```

Comment:
Can you add a cap to the backoff so that we don't exceed a reasonable limit? seems to me like it can exceed to an unreasonably high value

Reply:
> Can you add a cap to the backoff so that we don't exceed a reasonable limit? seems to me like it can exceed to an unreasonably high value

but don't we want to add a learner if control comes here then it means we are sure it is the scale-up case, so I guess better to retry with backoff. WDYT?  that's why I passed `retrySteps=6` . Do you have any other ideas/suggestions ?

Reply:
ok, I have set the upper limit of 120sec ?